sw/host/tests/chip/jtag/src/test_sram_load.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+use std::path::PathBuf;
+use std::fs;
+use std::str::FromStr;
+
+use regex::Regex;
+use anyhow::Result;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::{JtagParams, JtagTap, RiscvReg, RiscvCsr, RiscvGpr};
+use opentitanlib::util::vmem::Vmem;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::load_sram_program;
+use opentitanlib::uart::console::UartConsole;
+
+// use top_earlgrey::top_earlgrey_memory;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+    /// Path to the program to load
+    #[structopt(long)]
+    vmem: PathBuf,
+}
+
+fn test_sram_load(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    //
+    // Connect to the RISC-V TAP
+    //
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    let jtag = transport.jtag(&opts.jtag)?;
+    log::info!("Connecting to RISC-V TAP");
+    jtag.connect(JtagTap::RiscvTap)?;
+    log::info!("Halting core");
+    jtag.halt()?;
+    log::info!("Loading program to execute");
+    let vmem_content = fs::read_to_string(&opts.vmem)?;
+    let mut vmem = Vmem::from_str(&vmem_content)?;
+    // TODO don't hardcore this value, it depends on the linker script, there has
+    // to be a better way
+    let sram_load_addr = 0x10001fc4u32; // TODO don't hardcore that
+    let sram_stack_top = 0x10020000u32;
+    let sram_stack_size = 128;
+    let global_pointer = sram_load_addr + 2048;
+    log::info!("Uploading program to SRAM at {:x}", sram_load_addr);
+    vmem.merge_sections();
+    load_sram_program::load_vmem_program(&jtag, &vmem, sram_load_addr)?;
+    log::info!("Preparing ePMP for execution");
+    load_sram_program::prepare_epmp_for_sram(&jtag)?;
+    load_sram_program::setup_stack(&jtag, sram_stack_top, sram_stack_size)?;
+    // log::info!("set PC to {:x}", sram_load_addr);
+    // jtag.write_riscv_reg(&RiscvReg::CsrByName(RiscvCsr::DPC), sram_load_addr)?;
+    log::info!("set SP to {:x}", sram_stack_top);
+    jtag.write_riscv_reg(&RiscvReg::GprByName(RiscvGpr::SP), sram_stack_top)?;
+    log::info!("set GP to {:x}", global_pointer);
+    jtag.write_riscv_reg(&RiscvReg::GprByName(RiscvGpr::GP), global_pointer)?;
```

Comment:
it would be nice if we could move the SP/GP configuration to the `load_sram_progam` function

Reply:
perhaps same with the vmem merge sections and ePMP configuration operations? since the use case of this during manufacturing is to provide a way to inject SW into the chip before flash has been configured properly, it would be nice to have a single test_util that just loads a program ready to run

Reply:
Yes I will create a function that does everything.